### ISSUE TITLE
Fix FP S4123 (`no-invalid-await`): Don't flag functions returning promises and message

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#4516]
When calling `getTypeFromTreeNode()` on the node, the returned values has a `.symbol.escapedName` property that is equal to `Promise`

Reproducer:

```js
/**
 * @return {string}
 */
async function foo() {
  await fetch('https://unpkg.com/tailwindcss/dist/tailwind.min.css');
  return 'bar';
}

async function baz() {
  await foo(); // FP
}
```

Improve the message or the rule description: add `make sure that your function is returning a promise`

Proposal: Add `ts.TypeFlags.Void` to the white list

**Comment by ericmorand-sonarsource** [comment]
After further analysis, this appears to be more an _unexpected_ result than a false positive. When a JavaScript file is analyzed, the JSDoc `@return` directive is used to infer the called function's return type. Typically, the reproducible sample triggers the rule because the `foo` function is declared explicitly (through JSDoc `@return` directive) as returning a string instead of a Promise that resolves to a string.

No fix is actually needed. But since we will update the documentation to clarify the rule, we need to add some tests to assert that the JSDoc `@return` directive is honored as expected.

**Comment by ilia-kebets-sonarsource** [comment]
> When calling `getTypeFromTreeNode()` on the node, the returned values has a `.symbol.escapedName` property that is equal to `Promise`

We didn't manage to find any reproducible FP from the public issues that were reported, so we will not do this change now.

> Proposal: Add `ts.TypeFlags.Void` to the white list

We will not this since a function returning void is in fact not returning a promise.

**Comment by ericmorand-sonarsource** [comment]
The updated documentation looks good on Next.

**Comment by ilia-kebets-sonarsource** [comment]
new RSPEC PR: https://github.com/SonarSource/rspec/pull/3582
